pkg/prowgen/prowgen.go

```diff
@@ -312,7 +330,7 @@ func generatePodSpecMultiStage(info *ProwgenInfo, test *cioperatorapi.TestStepCo
 			Name: "ci-pull-credentials",
 		})
 	}
-	podSpec := generateCiOperatorPodSpec(info, secrets, []string{test.As})
+	podSpec := generateCiOperatorPodSpec(info, secrets, []string{test.As}, test.ClusterClaim != nil)
```

Comment:
What I meant with my other comment about the boolean flag is: At this point here in the code we know if we need the hive kubeconfig or not. So instead of adding another boolean flag, we can just append to the existing `secrets` arg